presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/LogsHandler.java

```diff
@@ -75,67 +109,75 @@ public static Response getLogs(String filename, JaxrsParameter<Instant> startDat
         requireNonNull(logLevel);
 
         if (!startDate.isValid() || !endDate.isValid()) {
-            return Response.status(Status.BAD_REQUEST)
-                    .entity("Invalid date format").build();
+            return badRequest("Invalid date format");
         }
         Instant start = startDate.get();
         Instant end = endDate.get();
 
-        if (start != null && end != null && maxEntries != null) {
-            return Response.status(Status.BAD_REQUEST)
-                    .entity("Can not provide date range and limit number of entries")
-                    .build();
+        if (start != null && end != null) {
+            if (maxEntries != null) {
+                return badRequest(
+                        "Can not provide date range and limit number of entries");
+            }
+            else if (start.isAfter(end)) {
+                return badRequest("End of date range can not be before start");
+            }
         }
 
         if (!logLevel.isValid()) {
-            return Response.status(Status.BAD_REQUEST)
-                    .entity("Invalid log level").build();
+            return badRequest("Invalid log level");
         }
         Level level = logLevel.get();
 
-        Path filePath = LOG_DIRECTORY.resolve(filename);
-        if (!Files.isRegularFile(filePath)) {
-            return Response.status(Status.NOT_FOUND)
-                    .entity(Files.exists(filePath)
-                            ? "Not a regular file"
-                            : "File not found")
-                    .build();
+        Path filePath;
+        try {
+            filePath = LOG_DIRECTORY.resolve(filename);
+        }
+        catch (InvalidPathException e) {
+            return badRequest("Invalid file name");
```

Comment:
Send it back (maybe truncated) and log the full thing -- much easier for the user